src/nuclio/common/components/edit-item/edit-item.component.js

```diff
@@ -100,6 +107,47 @@
                         };
                     })
                     .value();
+
+                ctrl.annotations = lodash.chain(ctrl.item.annotations)
+                    .defaultTo([])
+                    .map(function (value, key) {
+                        return {
+                            name: key,
+                            value: value,
+                            ui: {
+                                editModeActive: false,
+                                isFormValid: true,
+                                name: 'trigger.annotation'
+                            }
+                        };
+                    })
+                    .value();
+            }
+
+            if (isKafkaTrigger()) {
+                if (lodash.isNil(ctrl.item.attributes.initialOffset)) {
+                    ctrl.item.attributes.initialOffset = 'latest';
+                }
+
+                if (lodash.isNil(ctrl.item.attributes.sasl)) {
+                    ctrl.item.attributes.sasl = {};
+                }
```

Comment:
The following snippet can be instead of both of the above `if` statements:
```js
lodash.defaults(ctrl.item.attributes, {
    initialOffset: 'latest',
    sasl: {}
});
```
*UNLESS* `ctrl.item.attributes.sasl` could be `null`. `defaults` method works only with `undefined`.